romtime/src/mci.rs

```diff
@@ -57,4 +81,46 @@ impl Mci {
     pub fn disable_wdt(&self) {
         self.registers.mci_reg_wdt_timer1_en.set(0); // Timer1En CLEAR
     }
+
+    /// Read the reset reason register value
+    pub fn reset_reason(&self) -> u32 {
+        self.registers.mci_reg_reset_reason.get()
+    }
+
+    /// Get the reset reason as an enum
+    pub fn reset_reason_enum(&self) -> McuResetReason {
+        let reason = self.reset_reason();
+
+        let warm_reset = (reason & WARM_RESET_BIT) != 0;
```

Comment:
You should be able to use `.read()` to read these directly from the Tock register: https://github.com/chipsalliance/caliptra-mcu-sw/blob/main/registers/generated-firmware/src/mci.rs#L123

Reply:
Done.